tket/src/Transformations/Decomposition.cpp

```diff
@@ -746,6 +746,21 @@ Transform decomp_boxes() {
 
 Transform compose_phase_poly_boxes() {
   return Transform([](Circuit &circ) {
+    // replace wireswaps with three CX
+    while (circ.has_implicit_wireswaps()) {
+      bool foundswap = false;
+
+      qubit_map_t perm = circ.implicit_qubit_permutation();
+      for (const std::pair<const Qubit, Qubit> &pair : perm) {
+        if (pair.first != pair.second) {
+          if (!foundswap) {
+            circ.replace_implicit_wire_swap(pair.first, pair.second);
+            foundswap = true;
+          }
+        }
+      }
+    }
```

Comment:
Is the `foundswap` variable really necessary? Can you not do something like:
```suggestion
    while (circ.has_implicit_wireswaps()) {

      qubit_map_t perm = circ.implicit_qubit_permutation();
      for (const std::pair<const Qubit, Qubit> &pair : perm) {
        if (pair.first != pair.second) {
            circ.replace_implicit_wire_swap(pair.first, pair.second);
          }
        }
      }
    }
```

Reply:
Without the variable only one swap will create an infinity loop. The replace_implicit_wire_swap will remove one implicit wire swap and then add it again in the next iteration of the loop where pair.first != pair.second.

Reply:
Ah, okay. Would it be slightly faster then to have a `break` rather than keep iterating through the loop but not replacing the swaps?

Reply:
I have added that.